packages/lib/modules/pool/actions/add-liquidity/form/TokenInputs.tsx

```diff
@@ -31,9 +31,8 @@ export function TokenInputs({ tokenSelectDisclosureOpen, customSetAmountIn }: Pr
 
   function weightFor(tokenAddress: string) {
     return (
-      pool.poolTokens.find(token =>
-        isSameAddress(token.address as Address, tokenAddress as Address)
-      )?.weight ?? undefined
+      tokens.find(token => isSameAddress(token.address as Address, tokenAddress as Address))
+        ?.weight ?? undefined
```

Comment:
This works because in the `tokens` construction the high level token and it's underlying are merged so that the weight is attribute is preserved. `{ ...token, ...underlyingToken }`

Reply:
It could break for nested pools?